publication-commons/src/test/java/no/unit/nva/service/impl/DynamoDBPublicatonServiceTest.java

```diff
@@ -0,0 +1,172 @@
+package no.unit.nva.service.impl;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.amazonaws.services.dynamodbv2.local.embedded.DynamoDBEmbedded;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.unit.nva.Environment;
+import no.unit.nva.PublicationHandler;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Organization;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationStatus;
+import no.unit.nva.model.PublicationSummary;
+import org.junit.Rule;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.net.URI;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static no.unit.nva.service.impl.PublicationsDynamoDBLocal.BY_PUBLISHER_INDEX_NAME;
+import static no.unit.nva.service.impl.PublicationsDynamoDBLocal.NVA_RESOURCES_TABLE_NAME;
+import static no.unit.nva.service.impl.RestPublicationService.NOT_IMPLEMENTED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@EnableRuleMigrationSupport
+public class DynamoDBPublicatonServiceTest {
+
+    public static final String OWNER = "owner@example.org";
+    public static final URI PUBLISHER_ID = URI.create("http://example.org/123");
+    public static final String TABLE_NAME_ENV = "TABLE_NAME";
+    public static final String BY_PUBLISHER_INDEX_NAME_ENV = "BY_PUBLISHER_INDEX_NAME";
+    public static final String INVALID_JSON = "{\"test\" = \"invalid json }";
+
+    @Rule
+    public PublicationsDynamoDBLocal publicationsDynamoDBLocal =  new PublicationsDynamoDBLocal();
+
+    private ObjectMapper objectMapper = PublicationHandler.createObjectMapper();
+    private DynamoDBPublicationService publicationService;
+    private Environment environment;
+    private AmazonDynamoDB client;
+
+    /**
+     * Set up environment.
+     */
+    @BeforeEach
+    public void setUp() {
+        client = DynamoDBEmbedded.create().amazonDynamoDB();
+        environment = mock(Environment.class);
+        publicationService = new DynamoDBPublicationService(
+                objectMapper,
+                publicationsDynamoDBLocal.getByPublisherIndex()
+        );
+    }
+
+    @Test
+    @DisplayName("calling Constructor When Missing Env Throws Exception")
+    public void callingConstructorWhenMissingEnvThrowsException() {
+        assertThrows(IllegalStateException.class,
```

Comment:
if you do
`exception= assertThrows` you can get the exception and test that the message has the name of env variable